### PR TITLE
[feature]: Using the product as a copilot

### PR DESCRIPTION
**Pull request by allgandalf**
### Details

Add delegate avatar and delegate message
### Fixed Issues

$ https://github.com/Expensify/App/issues/46926
PROPOSAL: https://expensify.slack.com/archives/C02NK2DQWUX/p1725046112539189?thread_ts=1725040927.420699&cid=C02NK2DQWUX

### Tests

* Precondition: delegate should have full access to the delegator account

1. Login to delegate account.
2. Switch to delegator account.
3. Performs action like:
  - Commenting
  - Creating a Task, updating task, completing,  assigning task
  - Submitting an expense (Not creating but submitting ****)
  - Approving an expense
  - Paying an expense

* Verify that, all these actions have header `on behalf of`:
image

- [x] Verify that no errors appear in the JS console

### Offline tests

* Precondition: delegate should have full access to the delegator account

1. Login to delegate account.
2. Switch to delegator account.
3. Performs action like:
  - Commenting
  - Creating a Task, updating task, completing,  assigning task
  - Submitting an expense (Not creating but submitting ****)
  - Approving an expense
  - Paying an expense

* Verify that, all these actions have header `on behalf of`:
image

### QA Steps

* Precondition: delegate should have full access to the delegator account

1. Login to delegate account.
2. Switch to delegator account.
3. Performs action like:
  - Commenting
  - Creating a Task, updating task, completing,  assigning task
  - Submitting an expense (Not creating but submitting ****)
  - Approving an expense
  - Paying an expense

* Verify that, all these actions have header `on behalf of`:
image

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
Screenshot 2024-09-06 at 1 18 26 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/languages/en.ts
Comment (DylanDylann) [inline src/languages/en.ts:4603]: Add me as a reviewer

**Comment by DylanDylann** [comment]
@dangrous Should we wait for the BE change?

**Comment by dangrous** [comment]
BE should be up tomorrow 🤞🏻 , so I think focus on the optimistic / offline updates here and then we can do the final checks with what comes from the server. Everything should have a delegateAccountID in the action (I think, if not there than in the message) once that's complete.

**Comment by dangrous** [comment]
I think backend should mostly be live! There are a couple edge cases that might get wonky (mostly policy change logs) but I think you should be getting the `delegateAccountID` now - let me know!

**Comment by DylanDylann** [comment]
@dangrous I will try to complete the checklist soon

**Comment by DylanDylann** [comment]
From the design doc: 

Screenshot 2024-09-13 at 12 03 01


@allgandalf we should update the updated message in admin room 

https://github.com/user-attachments/assets/9676aa39-b178-4419-83f2-04f8e0fbd701


And the send invoice message
 
https://github.com/user-attachments/assets/0f0d5ba2-991c-420f-b933-7b31a073d468

**Comment by DylanDylann** [comment]
And also the submit expense flow

**Comment by DylanDylann** [comment]
Completing task message too 

Screenshot 2024-09-13 at 12 06 35

**Comment by DylanDylann** [comment]
@allgandalf @dangrous Oh wait, I want to confirm again the scope of this PR. Which actions should we implement in this PR? Or all actions as mentioned in the document

**Comment by dangrous** [comment]
so hopefully with the new backend changes, everything that's done as a delegate will always appropriately have the delegateAccountID attached to it - without actually needing to send anything new (for any action) from the front end. So I think basically every report action should show the "on behalf of" message, if there's a delegateAccountID. Is that helpful? I think there might have been a sentence missing in [your question](https://github.com/Expensify/App/pull/48594#issuecomment-2348040276) so not 100% sure I answered it right

**Comment by DylanDylann** [comment]
@allgandalf  Please give the list of actions that have already shown the "on behalf of" message, if It is triggered by the delegate account. And please tick which actions that are done in this PR

**Comment by dangrous** [comment]
Hey @allgandalf! So the backend change is still in process but will add it for task creation/edits/etc. too so let's add that to the optimistic data if we haven't yet. It'll disappear for now but I'm hoping that backend PR will be merged and ready tomorrow at the latest.

**Comment by dangrous** [comment]
Apologies on the delay - BE for `delegateAccountID` is still waiting on review, we're almost there though I think!

For the threads - I think maybe we leave that for now and I can try to get the backend updated. I think I know how to fix it, so I can test it quickly with your PR. Shouldn't block it though

**Comment by dangrous** [comment]
> Also whats the latest on system logs for policy changes ? that is also `BE`, I will get this PR ready for review today

This is the same PR! So same ETA.

**Comment by DylanDylann** [comment]
Thanks for bump, on my list today

**Comment by DylanDylann** [comment]
BUG: Don't block pay action when access limited

https://github.com/user-attachments/assets/1ccb1793-30be-4fa0-ae3e-343f1c2e477c

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (DylanDylann) [reply src/pages/home/report/ReportActionItemSingle.tsx:112]: @allgandalf Bump

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:4721]: Should we remove delegate params and get delegateID inside this function?

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:4290]: Should we remove delegate params and get delegateID inside this function?

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:4755]: Same above

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:4830]: Same above

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:4958]: same above

**Comment by DylanDylann** [comment]
BUG: There are no "on behalf of dylandylan9112+935@gmail.com" when submitting expenses offline 

https://github.com/user-attachments/assets/c57ab5ea-563c-4289-829f-5013063dd229

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:5005]: I don't see Track Expense actions are required for this feature. IMO, we should add this. But need confirmation to sure

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:5100]: Should we remove delegate params and get delegateID inside this function?

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:5454]: same above

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:5501]: same above

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [inline src/libs/ReportUtils.ts:6926]: same above

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (DylanDylann) [inline src/pages/home/report/ReportActionItemSingle.tsx:291]: should we use action?.delegateAccountID instead of delegateEmail

**Comment by DylanDylann** [comment]
BUG: There are no "on behalf of [dylandylan9112+935@gmail.com](mailto:dylandylan9112+935@gmail.com)" when paying expenses offline

https://github.com/user-attachments/assets/56079477-fbc7-4378-abd6-e37ca21d5fba

**Comment by DylanDylann** [comment]
BUG: The avatar display wrong (should address in another issue)

https://github.com/user-attachments/assets/b94756b8-0075-4da0-ab29-248346fab55a

**Comment by DylanDylann** [comment]
@allgandalf Reply https://github.com/Expensify/App/pull/48594#issuecomment-2370875975

I think we are missing `delegateID` in some optimisticData

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [reply src/libs/ReportUtils.ts:4290]: @allgandalf Yeah. But the value of this params isn't change. I don't see any reason why we should pass it as a param

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (DylanDylann) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: tbh, I don't see any reason why we need to use delegateEmail. There is no difference between using action?.delegateAccountID and delegateEmail. I suggest using delegateAccountID for more cleaner

Comment (rushatgabhane) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: delegateAccountID is only used for logs and stuff on the backend.

we should use `delegateEmail` by default

Comment (DylanDylann) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: @rushatgabhane In this case, delegateEmail is retrieved from delegateAccountID

```
    const delegatePersonalDetails = personalDetails[action?.delegateAccountID ?? ''];

    const delegateEmail = delegatePersonalDetails?.login;
```

Comment (rushatgabhane) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: oh lol mb!

File: src/libs/ReportUtils.ts
Comment (DylanDylann) [reply src/libs/ReportUtils.ts:4290]: I still see we are using Onyx.connect in reportUtils. On the other hand, we can create a new function to get delegateEmail and use in reportUtils

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (DylanDylann) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: @allgandalf Any thought on this comment?

File: src/libs/actions/IOU.ts
Comment (DylanDylann) [inline src/libs/actions/IOU.ts:7279]: remove redundant
```suggestion
```

File: src/libs/actions/IOU.ts
Comment (DylanDylann) [inline src/libs/actions/IOU.ts:7028]:
```suggestion
```

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (DylanDylann) [reply src/pages/home/report/ReportActionItemSingle.tsx:289]: @allgandalf One left comment here, everything else looks good

**Comment by DylanDylann** [comment]
@allgandalf Conflict

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]
LGTM

**Comment by DylanDylann** [comment]
@allgandalf Please merge main to solve Jest Unit Tests / test (job 2) (pull_request)

**Comment by rushatgabhane** [comment]
we definitely should QA test this

**Comment by dangrous** [comment]
Still waiting on BE unfortunately (it was two part, and the one part is still moving through staging/prod). i'm hoping tomorrow? But I've said that before haha... Repsonding to 

* Correct on the first couple points
* The invoice one is interesting - I will try to figure out if that's expected behavior; if it is we should disallow on the front end too.
* I was wondering about deleting the workspace too, I noticed that as well. I'll try to confirm if that (and changing the settings) is something we should handle now.

Will look at the PR shortly!

**Comment by dangrous** [comment]
Yep I'll double check but as long as any major issues are contained within the beta we should be fine to not revert. We should keep watch though once we merge to make sure we can fix anything that happens to be outside of the beta quickly!

**Comment by dangrous** [comment]
Re: that Slack conversation, sounds like no changes are needed on the front end for workspace settings access, and send invoices. I will need to raise a backend PR to allow for sending invoices but @allgandalf I don't think anything needs to change on the front end for that, right?

File: src/libs/ReportUtils.ts
Comment (dangrous) [inline src/libs/ReportUtils.ts:4181]:
```suggestion
    const delegateAccountDetails = PersonalDetailsUtils.getPersonalDetailByEmail(delegateEmail);

```

**Review by dangrous — CHANGES_REQUESTED** [review]
This looks good! One minor style comment, but then I think we're basically ready to merge EXCEPT:

The BE PR was FINALLY merged! In the next 12 hours or less it should be deployed; we should probably just give this another set of quick tests, but I bet it'll be good to go!

**Comment by dangrous** [comment]
Posting a couple bugs as I find them:
* I haven't figured out exact steps but creating an account, adding a copilot, immediately logging in as that copilot and switching accounts, then take an action, seems to know it's a copilot action but not showing the correct name. I checked Onyx and the `delegateAccountID` is there and correct:
Screenshot 2024-09-25 at 11 44 28
* In pretty much all cases I'm seeing a blank line where the original account should be when trying to switch back from a copiloted account.
Screenshot 2024-09-25 at 11 44 44

Let me know if you think either/both of these is backend but I don't think they are? The second might be the personal details thing we're working through separately.

**Comment by dangrous** [comment]
> > I haven't figured out exact steps but creating an account, adding a copilot, immediately logging in as that copilot and switching accounts, then take an action, seems to know it's a copilot action but not showing the correct name. I checked Onyx and the delegateAccountID is there and correct (all of these comments were sent from copilots, the bottom one is a fresh copilot I just added):
> 
> TBH i am really clueless how these system messages really work, I spend half day last week finding code for the messages and found out they come from `BE`, What i propose is, lets ship this PR and follow up on this issue on slack if it occurs (The system messages worked fine for me during testing now).

For this one it was happening for chats for me as well:

image

That's from an account copiloted in - so the display name and icon should be the copilot account, but it's the account that is being copiloted into...

I'm also just seeing the blank line on the basic flow of adding a copilot, logging in as the copilot, and switching into the copiloted account. 

However, I just tried on the staging server and it looks like these are okay there so it must be something with my local. Hopefully it's fine, though I'm not sure what the difference would be.

**Review by dangrous — APPROVED** [review]
let's do it!

**Comment by dangrous** [comment]
@allgandalf can you take a look at this [typeerror](https://github.com/Expensify/App/issues/49816):

`Type '{ common: { cancel: string; dismiss: string; yes: string; no: string; ok: string; notNow: string; learnMore: string; buttonConfirm: string; name: string; attachment: string; attachments: string; center: string; ... 224 more ...; rename: string; }; ... 173 more ...; debug: { ...; }; }' does not satisfy the expected type 'TranslationDeepObject<{ common: { cancel: string; dismiss: string; yes: string; no: string; ok: string; notNow: string; learnMore: string; buttonConfirm: string; name: string; attachment: string; attachments: string; center: string; ... 224 more ...; rename: string; }; ... 173 more ...; debug: { ...; }; }>'.
  The types of 'delegate.onBehalfOfMessage' are incompatible between these types.
    Type '(delegator: string) => string' is not assignable to type '(arg: Record<string, unknown>, ...noOtherArguments: unknown[]) => string | PluralForm'.
      Types of parameters 'delegator' and 'arg' are incompatible.
        Type 'Record<string, unknown>' is not assignable to type 'string'.`

**Comment by dangrous** [comment]
actually I think I have it, i can put up a PR in a sec

**Comment by dangrous** [comment]
https://github.com/Expensify/App/pull/49821 if @DylanDylann or @allgandalf you want to take a look

**Pull request merged**
